python/cudf/cudf/core/index.py

```diff
@@ -1714,7 +1714,13 @@ def __init__(
             raise TypeError("dtype must be a datetime type")
 
         name = _setdefault_name(data, name=name)["name"]
-        data = column.as_column(data, dtype=dtype)
+        data = column.as_column(data)
+
+        if isinstance(data.dtype, pd.DatetimeTZDtype):
```

Comment:
Why are we doing this after `as_column`? Will `as_column` not choke on this? I guess it just produces a column with the right dtype but the data isn't actually timezone-aware so we have to fail after?

Reply:
Before there appeared to be this special carve-out in `as_column` specifically for `DatetimeIndex` since it would call `as_column(..., dtype=datetime64[...])` and wanted to avoid a timezone-aware result.

```python
                if is_datetime64_dtype(dtype):
                    # Error checking only, actual construction happens
                    # below.
                    pa_array = pa.array(arbitrary)
                    if (
                        isinstance(pa_array.type, pa.TimestampType)
                        and pa_array.type.tz is not None
                    ):
                        raise NotImplementedError(
                            "cuDF does not yet support timezone-aware "
                            "datetimes"
                        )
```

Now this could actually be supported since @shwina added timezone support. I'll add a comment clarifying this